src/experimental/patterns/ec2-app.test.ts

```diff
@@ -0,0 +1,407 @@
+    template.hasResource("AWS::AutoScaling::AutoScalingGroup", {
+      Properties: {
+        HealthCheckGracePeriod: 120,
+      },
+      CreationPolicy: {
+        AutoScalingCreationPolicy: {
+          MinSuccessfulInstancesPercent: 100,
+        },
+        ResourceSignal: {
+          Count: 1,
+          Timeout: targetGroupHealthcheckTimeout.toIsoString(),
+        },
+      },
+    });
+  });
+
+  it("should add to the end of the user data", () => {
+    const stack = simpleGuStackForTesting();
+
+    const userDataCommand = `echo "Hello there"`;
+    const userData = UserData.forLinux();
+    userData.addCommands(userDataCommand);
+
+    const { autoScalingGroup } = new GuEc2AppExperimental(stack, { ...initialProps(stack), userData });
+
+    const renderedUserData = autoScalingGroup.userData.render();
+    const splitUserData = renderedUserData.split("\n");
+    const totalLines = splitUserData.length;
+
+    const appCommandPosition = splitUserData.indexOf(userDataCommand);
+    const startMarkerPosition = splitUserData.indexOf("# GuEc2AppExperimental UserData Start");
+    const endMarkerPosition = splitUserData.indexOf("# GuEc2AppExperimental UserData End");
+
+    // Application user data should be before the target group healthcheck polling.
+    expect(appCommandPosition).toBeLessThan(startMarkerPosition);
+
+    // The target group healthcheck polling should be the last thing in the user data.
+    expect(endMarkerPosition).toEqual(totalLines - 1);
+  });
+
+  it("should adjust properties of a horizontally scaling service", () => {
+    const cdkApp = new App();
+    const stack = new GuStack(cdkApp, "test", {
+      stack: "test-stack",
+      stage: "TEST",
+    });
+
+    const scalingApp = "my-scaling-app";
+    const { autoScalingGroup } = new GuEc2AppExperimental(stack, {
+      ...initialProps(stack, scalingApp),
+      scaling: {
+        minimumInstances: 5,
+      },
+    });
+    autoScalingGroup.scaleOnRequestCount("ScaleOnRequests", {
+      targetRequestsPerMinute: 100,
+    });
+
+    /*
+    We're ultimately testing an `Aspect`, which appear to run only at synth time.
+    As a work-around, synth the `App`, then perform assertions on the resulting template.
+
+    See also: https://github.com/aws/aws-cdk/issues/29047.
+     */
+    const { artifacts } = cdkApp.synth();
+    const cfnStack = artifacts.find((_): _ is CloudFormationStackArtifact => _ instanceof CloudFormationStackArtifact);
+
+    if (!cfnStack) {
+      throw new Error("Unable to locate a CloudFormationStackArtifact");
+    }
+
+    const template = Template.fromJSON(cfnStack.template as Record<string, unknown>);
+
+    const parameterName = `MinInstancesInServiceFor${scalingApp.replaceAll("-", "")}`;
+
+    template.hasParameter(parameterName, {
+      Type: "Number",
+      Default: 5,
+      MaxValue: 9, // (min * 2) - 1
+    });
+
+    template.hasResource("AWS::AutoScaling::AutoScalingGroup", {
```

Comment:
As your comment below mentions, I think we are really interested in the values of two properties here. Perhaps it would be clearer to have a separate test/test name for each?